packages/vuetify/src/components/VList/VListChildren.tsx

```diff
@@ -43,65 +45,53 @@ export const VListChildren = genericComponent<new <T extends InternalListItem>(
-        title: slots.title ? (slotProps: any) => slots.title?.({ ...slotProps, item }) : undefined,
-      }
+      return props.items?.map((internalItem, index) => {
+        const { children, props: itemProps, raw: item } = internalItem
 
-      const listGroupProps = VListGroup.filterProps(itemProps)
+        const slotsWithItem = {
+          subtitle: slots.subtitle ? (slotProps: any) => slots.subtitle?.({ ...slotProps, item }) : undefined,
+          prepend: slots.prepend ? (slotProps: any) => slots.prepend?.({ ...slotProps, item }) : undefined,
+          append: slots.append ? (slotProps: any) => slots.append?.({ ...slotProps, item }) : undefined,
+          title: slots.title ? (slotProps: any) => slots.title?.({ ...slotProps, item }) : undefined,
+        }
 
-      return children ? (
-        <VListGroup
-          value={ itemProps?.value }
-          { ...listGroupProps }
-        >
-          {{
-            activator: ({ props: activatorProps }) => {
-              const listItemProps = {
-                ...itemProps,
-                ...activatorProps,
-                value: props.returnObject ? item : itemProps.value,
-              }
+        const listGroupProps = VListGroup.filterProps(itemProps)
 
-              return slots.header
-                ? slots.header({ props: listItemProps })
-                : (
-                  <VListItem { ...listItemProps } v-slots={ slotsWithItem } />
-                )
-            },
-            default: () => (
-              <VListChildren
-                items={ children }
-                returnObject={ props.returnObject }
-                v-slots={ slots }
-              />
-            ),
-          }}
-        </VListGroup>
-      ) : (
-        slots.item ? slots.item({ props: itemProps }) : (
-          <VListItem
-            { ...itemProps }
-            value={ props.returnObject ? item : itemProps.value }
-            v-slots={ slotsWithItem }
-          />
+        return children && itemProps.subheader ? (
+          <div>
+            <VListSubheader { ...itemProps }></VListSubheader>
+            <VListChildren items={ children } v-slots={ slots } />
+            { itemProps.divider && <VDivider /> }
+          </div>
+        ) : children ? (
+          <VListGroup
+            value={ itemProps?.value }
+            { ...listGroupProps }
+          >
+            {{
+              activator: ({ props: activatorProps }) => slots.header
+                ? slots.header({ item: internalItem, props: { ...itemProps, ...activatorProps }, index })
+                : <VListItem { ...itemProps } { ...activatorProps } v-slots={ slotsWithItem } />,
+              default: () => (
+                <VListChildren
+                  items={ children }
+                  v-slots={ slots }
+                />
+              ),
+            }}
+          </VListGroup>
+        ) : (
+          slots.item ? slots.item({ item: internalItem, props: itemProps, index }) : (
```

Comment:
is this kind of change breaking or do you not need to use ALL available properties when redefining slots?

Reply:
My question was about the definition of a custom slot.
So for example, if user has this in their template:

```
<template #item="{ props }">

</template>
```

Would adding new available parameters for the slot here, break their code.

But since we are not changing the naming of `props`, it should work fine :)